### PR TITLE
NumFOCUS updates

### PR DESCRIPTION
**Pull request by bryanwweber**
Changes proposed in this pull request:
- Add NumFOCUS information to docs pages
- Remove custom Mathjax extension because one is available from Sphinx
- Update copyright year

Any comments/updates/text suggestions are welcome!

### CODE REVIEW

File: doc/sphinx/about.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+.. _sec-about:
+
+*************
+About Cantera
+*************
+
+Cantera is an open source scientific computing project, that allows users to
+quickly and efficiently incorporate and execute fundamental thermodynamic,
+chemical kinetic, and species transport calculations in their work. This can be
+achieved by standalone command-line calls to core Cantera functions,
```
Comment (speth) [inline doc/sphinx/about.rst]: I'm not sure I understand what is meant by "command-line calls", at least as far as what Cantera currently has.

File: doc/sphinx/about.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+incorporating these calls into bespoke code or software, or by using any of the
+solvers or examples distributed with the Cantera software package.
+
+Cantera was originally started by Prof. David G. Goodwin at the California
+Institute of Technology. Building on Prof. Goodwin's legacy, Cantera is licensed
+under a permissive 3-Clause BSD license, ensuring that the software will remain
```
Comment (speth) [inline doc/sphinx/about.rst]: We could link directly to the license here.

File: doc/sphinx/about.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+The role of the steering committee is to ensure the long-term health of the
+Cantera project. This includes overseeing the development of the Cantera code
+and community in a way that most benefits the entire community of users and
+contributors. The steering committee can be contacted at `steering@cantera.org
+<mailto:steering@cantera.org>`_ The current steering committee (in alphabetical
+order) is:
+
+    * Steven DeCaluwe, Colorado School of Mines
+    * \C. Franklin Goldsmith, Brown University
```
Comment (speth) [inline doc/sphinx/about.rst]: What does the backslash do here?

**Review by speth — APPROVED** [review]
Overall, this looks good to me, Bryan. Merge whenever you're ready.

File: doc/sphinx/about.rst (hunk 3 above)
Comment (bryanwweber) [reply doc/sphinx/about.rst]: Prevents the `C.` from being interpreted as an ordered sublist

File: doc/sphinx/about.rst (hunk 1 above)
Comment (bryanwweber) [reply doc/sphinx/about.rst]: How about `achieved by interactive calls to core...`?

File: doc/sphinx/about.rst (hunk 2 above)
Comment (bryanwweber) [reply doc/sphinx/about.rst]: :+1:

**Pull request closed without merging**